webgpu.h

```diff
@@ -1295,77 +1373,200 @@ typedef struct WGPUStorageTextureBindingLayout {
+     */
     size_t formatCount;
     WGPUTextureFormat const * formats;
+    /**
+     * A list of supported @ref WGPUPresentMode values.
+     * Guaranteed to contain @ref WGPUPresentMode_Fifo.
+     */
     size_t presentModeCount;
     WGPUPresentMode const * presentModes;
+    /**
+     * A list of supported @ref WGPUCompositeAlphaMode values.
+     * @ref WGPUCompositeAlphaMode_Auto will be an alias for the first element and will never be present in this array.
+     */
     size_t alphaModeCount;
     WGPUCompositeAlphaMode const * alphaModes;
 } WGPUSurfaceCapabilities WGPU_STRUCTURE_ATTRIBUTE;
 
+/**
+ * Options to `::wgpuSurfaceConfigure` for defining how a @ref WGPUSurface will be rendered to and presented to the user.
+ * See @ref Surface-Configuration for more details.
+ */
 typedef struct WGPUSurfaceConfiguration {
     WGPUChainedStruct const * nextInChain;
+    /**
+     * The @ref WGPUDevice to use to render to surface's textures.
+     */
     WGPUDevice device;
+    /**
+     * The @ref WGPUTextureFormat of the surface's textures.
+     */
     WGPUTextureFormat format;
+    /**
+     * The @ref WGPUTextureUsage of the surface's textures.
+     */
     WGPUTextureUsageFlags usage;
+    /**
+     * The width of the surface's textures.
+     */
+    uint32_t width;
+    /**
+     * The height of the surface's textures.
+     */
+    uint32_t height;
+    /**
+     * The additional @ref WGPUTextureFormat for @ref WGPUTextureView format reinterpretation of the surface's textures.
+     */
     size_t viewFormatCount;
     WGPUTextureFormat const * viewFormats;
+    /**
+     * How the surface's frames will be composited on the screen.
+     */
     WGPUCompositeAlphaMode alphaMode;
-    uint32_t width;
-    uint32_t height;
+    /**
+     * When and in which order the surface's frames will be shown on the screen.
+     */
     WGPUPresentMode presentMode;
 } WGPUSurfaceConfiguration WGPU_STRUCTURE_ATTRIBUTE;
 
+/**
+ * The root descriptor for the creation of an @ref WGPUSurface with `::wgpuInstanceCreateSurface`.
+ * It isn't sufficient by itself and must have one of the `WGPUSurfaceSource*` in its chain.
+ * See @ref Surface-Creation for more details.
+ */
 typedef struct WGPUSurfaceDescriptor {
     WGPUChainedStruct const * nextInChain;
+    /**
+     * Label used to refer to the object.
+     */
     WGPU_NULLABLE char const * label;
 } WGPUSurfaceDescriptor WGPU_STRUCTURE_ATTRIBUTE;
 
+/**
+ * Chained in @ref WGPUSurfaceDescriptor to make an @ref WGPUSurface wrapping an Android [`ANativeWindow`](https://developer.android.com/ndk/reference/group/a-native-window).
+ */
 typedef struct WGPUSurfaceSourceAndroidNativeWindow {
     WGPUChainedStruct chain;
+    /**
+     * The pointer to the [`ANativeWindow`](https://developer.android.com/ndk/reference/group/a-native-window) that will be wrapped by the @ref WGPUSurface.
+     */
     void * window;
 } WGPUSurfaceSourceAndroidNativeWindow WGPU_STRUCTURE_ATTRIBUTE;
 
+/**
```

Comment:
This will conflict with https://github.com/webgpu-native/webgpu-headers/pull/320